internal/provider/provider.go

```diff
@@ -15,22 +16,52 @@ func Provider() *schema.Provider {
 	return &schema.Provider{
 		Schema: map[string]*schema.Schema{
 			"sddc_manager_username": {
-				Type:        schema.TypeString,
-				Required:    true,
-				Description: "Username to authenticate to SDDC Manager",
-				DefaultFunc: schema.EnvDefaultFunc(constants.VcfTestUsername, nil),
+				Type:          schema.TypeString,
+				Optional:      true,
+				Description:   "Username to authenticate to SDDC Manager",
+				ConflictsWith: []string{"cloud_builder_username", "cloud_builder_password", "cloud_builder_host"},
```

Comment:
Yep. This basically means the provider has two modes of operaton. I've commented this with Amit: the expectation is that after the SDDCManager is up, there is no more need to call CloudBuilder APIs (users are likely to power off that VM). Hence, no more need to keep two API clients ready in the provider. 

Is there any meaningful info we can retrieve on the vcf_instance read, that can be used with the SDDCManager API?

Reply:
After SDDC Manager is fully deployed, you should be able to use this API for some basic info - https://developer.vmware.com/apis/vcf/latest/v1/sddc-managers/get/

On Cloud Builder, I believe only these apis are available: 

To get the bringup task details:
https://developer.vmware.com/apis/vcf/latest/v1/sddcs/get/
https://developer.vmware.com/apis/vcf/latest/v1/sddcs/id/get/

To get the SDDC Manager VM details:
https://developer.vmware.com/apis/vcf/latest/v1/sddcs/id/sddc-manager/get/

Reply:
Included the "sddc_manager_id", "sddc_manager_fqdn", "sddc_manager_version" as computed properties and populating them as part of the read operation. Left out the "sddc_manager_ip" because it is actually an input to the "create" of the "instance" resource.